### PR TITLE
take signable to the level really near web3j

### PR DESCRIPTION
**Pull request by colourful-land**
target being  `signable-merged-into-activity-reference-branch` so that only changes the changes in stage 2 signable is visible for easier review @JamesSmartCell - I'm still testing the result

This solves https://github.com/AlphaWallet/alpha-wallet-android/issues/1556

### CODE REVIEW

File: app/src/main/java/com/alphawallet/app/viewmodel/DappBrowserViewModel.java [hunk 1]
```diff
@@ -174,8 +175,8 @@ private void checkBalance(final Wallet wallet)
                             .find()).toObservable();
     }
 
-    public void signMessage(byte[] signRequest, DAppFunction dAppFunction, EthereumMessage message) {
-        disposable = createTransactionInteract.sign(defaultWallet.getValue(), signRequest, defaultNetwork.getValue().chainId)
+    public void signMessage(DAppFunction dAppFunction, Signable message) {
```
Comment (colourful-land) [inline app/src/main/java/com/alphawallet/app/viewmodel/DappBrowserViewModel.java:178]: Order of params is unconventional here.

File: app/src/main/java/com/alphawallet/app/viewmodel/WalletConnectViewModel.java [hunk 2]
```diff
@@ -66,8 +66,8 @@ public void getAuthenticationForSignature(Wallet wallet, Activity activity, Sign
         keyService.getAuthenticationForSignature(wallet, activity, callback);
     }
 
-    public void signMessage(byte[] signRequest, DAppFunction dAppFunction, Message<String> message) {
-        disposable = createTransactionInteract.sign(defaultWallet.getValue(), signRequest, 1)
+    public void signMessage(byte[] signRequest, DAppFunction dAppFunction, EthereumMessage message) {
```
Comment (colourful-land) [inline app/src/main/java/com/alphawallet/app/viewmodel/WalletConnectViewModel.java:69]: should reduce the params again here.

File: app/src/main/java/com/alphawallet/app/service/KeystoreAccountService.java [hunk 3]
```diff
@@ -325,9 +326,19 @@ public static Credentials getCredentials(File keyFolder, String address, String
         }).subscribeOn(Schedulers.io());
     }
 
+
     //In all cases where we need to sign data the signature needs to be in Ethereum format
     //Geth gives us the pure EC function, but for hash signing
-    @Override
```
Comment (colourful-land) [inline app/src/main/java/com/alphawallet/app/service/KeystoreAccountService.java:330]: Should still override one of the functions

**Comment by colourful-land** [comment]
@JamesSmartCell said:
> Have you tested this signing transactions from dapp? Places to check are:
> 
> http://thetokenfactory.com
> https://tf.alphawallet.com

The one I tested are:

1. Signing in to localcryptos.com

2. Deploy a contract using  https://tf.alphawallet.com (Ropstern)

Both are successful.

Therefore this one is ready to merge

**Comment by JamesSmartCell** [comment]
Made some modifications to the Signable class to support Typed messages, and wired up WalletConnect.

**Pull request merged**